pages/_middleware.ts

```diff
@@ -1,15 +1,31 @@
 import { NextMiddleware, NextRequest, NextResponse } from 'next/server';
+import { log } from '@/lib/log';
 
 // Next.js middleware
 // ==================
 // https://nextjs.org/docs/middleware
 
 export const middleware: NextMiddleware = ( req: NextRequest ) => {
+	const pathName = req.nextUrl.pathname;
+
 	// Required health check endpoint on VIP. Do not remove.
-	if ( req.nextUrl.pathname === '/cache-healthcheck' ) {
+	if ( pathName === '/cache-healthcheck' ) {
 		return new NextResponse( 'Ok' );
 	}
 
-	// Returning nothing means the request will continue as normal through the
-	// Next.js lifecycle.
+	// Ensures that when this response is returned, the Next.js chain will continue
+	const response = NextResponse.next();
+
+	if (req.headers.has( 'x-request-id' ) ) {
+		const sourceId = req.headers.get( 'x-request-id' );
```

Comment:
Should we store `x-request-id` as a variable, as it is used in more than one place?